uaclient/config.py

```diff
@@ -160,14 +161,29 @@ def data_path(self, key=None, private=True):
             return os.path.join(data_dir, self.data_paths[key])
         return os.path.join(data_dir, key)
 
+    def delete_cache_key(self, key):
+        """Remove specific cache file."""
+        if not key:
+            return
```

Comment:
Are there any cases where we would intentionally call `delete_cache_key` with a False-ish value? If not, then hitting this `return` implies that we have introduced a bug; could we either raise an exception or at least log an error here?

Reply:
Dan, correct, we shouldn't ever call this with a non-string or empty key. I've dropped that early exit and we can expect a RuntimeError or TypeError if we get invalid values passed in. Let's break loudly if this happens as it is a programming error.